### PR TITLE
3.0 - Article "Configure the saml-based SSO"

### PR DESCRIPTION
**Pull request by bgouraud**

### CODE REVIEW

File: content/md/user-management/configure-saml-sso.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+# Overview
+
+Single Sign-On (aka SSO) is an authentication process that allows a user to access multiple applications with only one set of credentials.
+
+Single Sign-On is a common practice around corporations that handles lots of users and lots of applications.
+
+## Common SAML terms
+
+Understanding how SAML-based Single Sign-On work can be a complicated task as it introduces terms that can be obscure at first glance.
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]:
```suggestion
Understanding how SAML-based Single Sign-On works can be a complicated task as it introduces terms that can be obscure at first glance.
```

File: content/md/user-management/configure-saml-sso.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+The SP metadata information can be accessed at the following URL: https://YOUR_PIM_SERVER/saml/metadata. Depending on your IdP, this metadata URL can be imported into the IdP to ease the configuration process of your application.
+:::
+
+* `Identity Provider`: The IdP can also supply metadata to help configure the SAML communication. At this time, it is not possible to import it in the PIM to configure the Single Sign-On.
+
+## What is SAML?
+
+SAML (Security Assertion Markup Language) is an XML-based standard for exchanging authentication data between an `Identity Provider (IdP)` (such as Microsoft Azure AD) and a `Service Provider (SP)` (in our case our beloved PIM).
+
+# Gather information from your Identity Provider (IdP)
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: It's a bit weird to have a section only for this. It feels empty. As this is the first step of the set up, maybe put it into the next section as subsections.

File: content/md/user-management/configure-saml-sso.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+# Configure your Service Provider (SP) in your PIM to communicate with the IdP
+
+## Activate/Disable the Single Sign-On feature
+
+### Activate the Single Sign-On feature
+
+To activate the Single Sign-On feature, just click on the `SSO enabled` button in the interface.
+Once enabled, all the authentication requests will be redirected to the authentication server you specified.
+
+### Disable the Single Sign-On feature
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: I wouldn't put this here as the title of the main section is more about configuring not disabling. I would put it that in a dedicated section afterward. What do you think?

File: content/md/user-management/configure-saml-sso.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+To disable the Single Sign-On feature, just click on the `SSO enabled` button in the interface.
+Once disabled, all the authentication requests will go through the classical login interface.
+
+## Configure the Identity Provider (IdP) section
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: A screenshot of the conf page here maybe?

File: content/md/user-management/configure-saml-sso.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* `Entity ID`: String that uniquely identify your IdP (it is in general provided by your IdP)
+* `Sign-on URL`: URL that will be used when a user tries to log in the PIM
+* `Logout URL`: URL that will be used when a user ask for logout action in the PIM
+* `Public certificate`: As the communications are encrypted between the SP and the IdP, the IdP public key is required in order to decrypt any incoming message.
+
+:::info
+Certificates must be in X.509 format.
+:::
+
+## Configure the Service Provider (SP) section
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: A screenshot here too?

File: content/md/user-management/configure-saml-sso.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+Using the information you gather previously, you can fill all the required fields that are required to communicate with the IdP.
+The mandatory values are the following:
+* `Entity ID`: String that uniquely identify your IdP (it is in general provided by your IdP)
+* `Sign-on URL`: URL that will be used when a user tries to log in the PIM
+* `Logout URL`: URL that will be used when a user ask for logout action in the PIM
+* `Public certificate`: As the communications are encrypted between the SP and the IdP, the IdP public key is required in order to decrypt any incoming message.
+
+:::info
+Certificates must be in X.509 format.
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: Can you find on the internet a documentation of this format? To add it as a link here.

File: content/md/user-management/configure-saml-sso.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+Once enabled, all the authentication requests will be redirected to the authentication server you specified.
+
+### Disable the Single Sign-On feature
+
+To disable the Single Sign-On feature, just click on the `SSO enabled` button in the interface.
+Once disabled, all the authentication requests will go through the classical login interface.
+
+## Configure the Identity Provider (IdP) section
+
+Using the information you gather previously, you can fill all the required fields that are required to communicate with the IdP.
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]:
```suggestion
Using the information you gathered previously, you can fill all the required fields that are required to communicate with the IdP.
```
Maybe you can put the previous section right here in fact.

File: content/md/user-management/configure-saml-sso.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+* `Logout URL`: URL that will be used when a user ask for logout action in the PIM
+* `Public certificate`: As the communications are encrypted between the SP and the IdP, the IdP public key is required in order to decrypt any incoming message.
+
+:::info
+Certificates must be in X.509 format.
+:::
+
+## Configure the Service Provider (SP) section
+
+The information required for configuring the SP are the following:
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: Would it be a bit clearer if you mention here that that information is populated by the PIM and should be used in the IdP (if I understood well).

File: content/md/user-management/configure-saml-sso.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+:::info
+By default, the SP configuration is automatically populated during the first access to the SSO configuration page. If you want to provide your own certificate, you can paste the public and private key certificates in the corresponding fields.
+:::
+
+:::info
+The SP certificate is self-signed and is generated to never expire. You will never have to renew it.
+:::
+
+# Tips and tricks for configuring the SAML-based SSO
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]:
```suggestion
# Tips and tricks
```

File: content/md/user-management/configure-saml-sso.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+## What is SAML?
+
+SAML (Security Assertion Markup Language) is an XML-based standard for exchanging authentication data between an `Identity Provider (IdP)` (such as Microsoft Azure AD) and a `Service Provider (SP)` (in our case our beloved PIM).
+
+# Gather information from your Identity Provider (IdP)
+
+First of all, before configuring your PIM to communicate with an `Identity Provider`, you will have to gather some information about your IdP. Your IT department/SSO administrator must probably have all the information.
+
+# Configure your Service Provider (SP) in your PIM to communicate with the IdP
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: I would simplify this title which is a bit too long.

```suggestion
# Configure your Service Provider (SP)
```

Comment (NolwennP) [reply content/md/user-management/configure-saml-sso.md]: Maybe add a sentence afterward to tell more.

File: content/md/user-management/configure-saml-sso.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+Once the SSO will be activated, the passwords that exists on the PIM will no longer be used.
+On the other hand, the users created in the PIM will still be needed to make the connection between the users coming from the IdP and the user rights existing in the PIM.
+
+You will have to make sure that all the potential users in your IdP that can access the PIM have an account configured in the PIM.
+Unknown users are not generated on-the-fly when accessing the PIM. If a user doesn't exist in the PIM, he will not be able to access the PIM and an error will be raised.
+
+## About user claims
+
+User claims are information that the Service Provider expects when a authentication is made on the IdP and the positive response comes back to the PIM.
+The IdP response must contain the following attributes:
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: Is it something the integrator should make sure to make appear in the IdP response? if so, I think you can state it very explicitly.

File: content/md/user-management/configure-saml-sso.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+* **akeneo-uid**: Will be used to check the username in the PIM database. This is the most important part of the SAML communication.
+* **http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname**: Last name of the user
+* **http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname**: First name of the user
+* **http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress**: Email of the user
+
+## About security
+
+By design, in order to have the more secured process, all communications between the Identity Provider and the Service Provider have to be encrypted.
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]:
```suggestion
By design, in order to have the most secure process, all communications between the Identity Provider and the Service Provider have to be encrypted.
```

Not sure about this one.

File: content/md/user-management/configure-saml-sso.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+Moreover, as far as security is involved, it is mandatory to use the HTTPS protocol around the PIM. Using HTTP communication may be a serious security breach.
```
Comment (NolwennP) [inline content/md/user-management/configure-saml-sso.md]: Same remark as before. I think you need to really say something like "Please make sure that you use...". Don't hesitate to give the perfect example you gave the other day when we were preparing the 3.0 presentation (don't remember it, but it was good :) ).

File: content/md/user-management/manage-your-users.md [hunk 14]
```diff
@@ -63,7 +63,7 @@ Even if a user is deleted, its actions generated in the PIM won't be erased.
 
 # Deal with the user permissions
 
-To know more about user permissions, there are four articles just for you depending on what your need are: 
+To know more about user permissions, there are four articles just for you depending on what your need are:
```
Comment (NolwennP) [inline content/md/user-management/manage-your-users.md]: Starting this point you already have these changes in another PR. I can help you with Git to remove them from here.

File: content/md/user-management/configure-saml-sso.md [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+For the `Service Provider`, the Entity ID is automatically generated and corresponds by default to the metadata URL of the SP.
+
+### Assertion Consumer Service (ACS)
+
+The ACS is the SP endpoint (URL) that is responsible for receiving the SAML response from the IdP.
+
+### Metadata
+
+Metadata is a set of information supplied either by the IdP or the SP, in XML format.
+* `Service Provider`: The SP supplied metadata provide the Entity ID, the X.509 certificate used for decrypting messages coming from the SP, the Logout URL and the ACS URL.
```
[the remark this answers is not in the document]
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: I rephrased this part to make it clearer (I hope).

File: content/md/user-management/configure-saml-sso.md (hunk 4 above)
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: I added a screenshot here.

File: content/md/user-management/configure-saml-sso.md (hunk 5 above)
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: I also added a screenshot here.

File: content/md/user-management/configure-saml-sso.md (hunk 6 above)
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: Yes. I added a link to the X.509 certificate format.

File: content/md/user-management/configure-saml-sso.md (hunk 10 above)
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: Reworked this part.

File: content/md/user-management/configure-saml-sso.md (hunk 11 above)
Comment (bgouraud) [reply content/md/user-management/configure-saml-sso.md]: Yep. Reworked this part to make it clearer.

**Comment by bgouraud** [comment]
I changed the two titles with 'Configure the SSO' and 'Disable the SSO'.

**Pull request merged**